content/docs/architecture/auth.md

```diff
@@ -0,0 +1,133 @@
+
+```json
+{
+    "Root": {
+        "CustomSpu": [
+            "All"
+        ],
+        "Partition": [
+            "All"
+        ],
+        "Spu": [
+            "All"
+        ],
+        "Topic": [
+            "All"
+        ],
+        "SpuGroup": [
+            "All"
+        ],
+        "ManagedConnector": [
+            "All"
+        ],
+        "SmartModule": [
+            "All"
+        ],
+        "TableFormat": [
+            "All"
+        ]
+    },
+    "TopicUser": {
+        "Topic": [
+            "Read"
+        ]
+    }
+
+}
+```
+
+Note that, in order to admin the cluster, "Root" role with full access to all objects is required.
+
+## Authorization Scope file
+
+The scope file contains mapping of users to roles. For example, here is a sample scope file that maps user `user1` and `user2` to `TopicUser` role.  The `Root` role is mapped to user `root`.
+
+```json
+{
+    "Root": [
+        "root"
+    ],
+    "TopicUser": [
+        "user1",
+        "user2"
+    ]
+}
+```
+
+The user id comes from the X509 client certs.  It is mapped to `CN`.  For example, this is a simple command to generate a client cert with `CN` set to `user1`.
+
+```bash
+openssl req -new -key <client_key> -out <client.csr> -subj "/C=US/ST=CA/O=MyOrg, Inc./CN=user1"
+```
+
+The client cert with `CN` set to `user1` can be used to connect to Fluvio cluster.  The Fluvio cluster will map the `user1` to `TopicUser` role.
+
+## Provisioning Fluvio cluster with TLS
+
+Once all the certs and keys are ready, you can provision a Fluvio cluster with TLS enabled.  Note that client certs must have user "root" (default policy mapping) in order to provision the cluster.  Here is syntax to provision a Fluvio cluster with TLS enabled.
+
+```bash
+fluvio cluster start --tls --domain <domain> --ca-cert ca_cert --server-cert <server_cert> --server-key <server_key> --client-cert <client_cert> --client-key <client_key> --authorization-policy <policy_file> --authorization-scope <scope_file>
+```
+
+The domain must match DNS suffice in order to TLS to work properly.  For example, if you are using self-signed certs, you can use `local` or `localhost` as domain.  If you are using certs signed by a trusted CA, you can use the domain name that is registered with the CA.
+
+# Connecting to Fluvio cluster with TLS
+
+In order to connect to Fluvio cluster with TLS enabled, you must provide client certs and keys. They must be signed by the same CA as the server cert and key.  In addition, it must have CN(user) that is in the authorization scope file.  The client TLS certs and key can be either provided in the fluvio profile file or specified in the env variables.  Here is an section of fluvio profile file that contains client certs and keys.   Currently TLS certs must be manually copied to config file.
```

Comment:
this section seems to be repeated.

Reply:
Different client can have different certs.